lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -1148,6 +1149,89 @@ spec = do
               txs2 <- listTransactions @n ctx w (Just te) (Just te) Nothing
               length <$> [txs1, txs2] `shouldSatisfy` all (== 0)
 
+    it "TRANS_GET_01 - Can get Incoming and Outgoing transaction" $ \ctx -> do
+        (wSrc, wDest) <- (,) <$> fixtureWallet ctx <*> emptyWallet ctx
+        -- post tx
+        let amt = (1 :: Natural)
+        rMkTx <- postTx ctx
+            (wSrc, Link.createTransaction @'Shelley, "cardano-wallet")
+            wDest
+            amt
+        let txid = getFromResponse #id rMkTx
+        verify rMkTx
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            , expectField (#status . #getApiT) (`shouldBe` Pending)
+            ]
+
+        -- Verify Tx in source wallet is Outgoing and Pending
+        let linkSrc = Link.getTransaction @'Shelley wSrc (ApiTxId txid)
+        r1 <- request @(ApiTransaction n) ctx linkSrc Default Empty
+        verify r1
+            [ expectResponseCode HTTP.status200
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            , expectField (#status . #getApiT) (`shouldBe` Pending)
+            ]
```

Comment:
Note that this is subject to race-condition, especially with a slot-length of 0.2s. There are good chances that the transaction is already inserted at this stage. I'd remove this assertion and the next one about the wallet balance (which is collateral to this test), and simply keep the last two assertions of this scenario behind an `eventually`. 

These would be sufficient for this test to show that transactions can be obtained using their ID :+1:

Reply:
agree here. this poses potential race condition - corrected in https://github.com/input-output-hk/cardano-wallet/pull/1795/commits/d898b0d84bba687166dcd71bfd440d29cd26f063